test/resources/knativeserving.go

```diff
@@ -21,68 +21,88 @@ package resources
 import (
 	"context"
 	"fmt"
+	"testing"
 	"time"
 
 	"github.com/pkg/errors"
 	v1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	va1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"knative.dev/pkg/test/logging"
 	"knative.dev/serving-operator/pkg/apis/serving/v1alpha1"
+	servingv1alpha1 "knative.dev/serving-operator/pkg/client/clientset/versioned/typed/serving/v1alpha1"
 	"knative.dev/serving-operator/test"
 )
 
 const (
-	interval = 10 * time.Second
-	timeout  = 2 * time.Minute
+	// Interval specifies the time between two polls.
+	Interval = 10 * time.Second
+	// Timeout specifies the timeout for the function PollImmediate to reach a certain status.
+	Timeout  = 5 * time.Minute
 )
 
-// WaitForKnativeServingState polls the status of the KnativeServing called name
+// WaitKnativeServingReady polls the status of the KnativeServing called name
 // from client every `interval` until `inState` returns `true` indicating it
 // is done, returns an error or timeout.
-func WaitForKnativeServingState(client *test.KnativeServingAlphaClients, name string, inState func(s *v1alpha1.KnativeServing) (bool, error)) (*v1alpha1.KnativeServing, error) {
+func WaitKnativeServingReady(t *testing.T, clients servingv1alpha1.KnativeServingInterface, name string,
```

Comment:
Why rename this? You still have an inState function there that might check for different things than "ready"?

Reply:
The reason was I was not sure about the status during deletion, also verified the case of deletion. Since we do not need to verify for delete, I can change it back.